src/linux/py_proc.h

```diff
@@ -68,6 +68,21 @@ union {
 
 
 // ----------------------------------------------------------------------------
+static Elf64_Addr
+_py_proc_get_base64(Elf64_Ehdr * ehdr, void * elf_map)
+{
+  Elf64_Addr res = (Elf64_Addr) 0 - 1;
+  for (uint16_t i = 0; i < ehdr->e_phnum; ++i) {
+    Elf64_Phdr * phdr = (Elf64_Phdr *) (elf_map + i * ehdr->e_phoff);
+    if (phdr->p_type == PT_LOAD) {
+      Elf64_Addr value = phdr->p_vaddr - phdr->p_vaddr % phdr->p_align;
```

Comment:
Please code as
~~~ c
Elf64_Addr value = phdr->p_vaddr & (-phdr->p_align);
~~~

Reply:
Wouldn't `Elf64_Addr value = phdr->p_vaddr & (~phdr->p_align);` be more portable ? (for non-one-complement CPUs)

Reply:
I'd say this implementtation has the drawbacks of being less clear and not compatible with non two-complement CPUs.

Reply:
Whilst I agree about being less clear, it has the advantage of being faster than the `%` operation. My bet is that Austin will never be compiled on a one's complement architecture, but one can never know :). In that case, perhaps one could be explicit and use instead
~~~ C
Elf64_Addr value = phdr->p_vaddr & (~phdr->p_align + 1);  // Round VA from below to multiple of page size.
~~~